include/one_dimensional_linear_interpolator.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef __ONE_DIMENSIONAL_LINEAR_INTERPOLATOR_H__
+#define __ONE_DIMENSIONAL_LINEAR_INTERPOLATOR_H__
+
+#include "interpolator.hpp"
+#include <CL/sycl.hpp>
+#include <mpi.h>
+#include <neso_particles.hpp>
+#include <vector>
+
+using namespace NESO::Particles;
+using namespace cl;
+namespace NESO {
+
+/**
+ *  Class used to output a vector of y values, given some x values, based
+ * on provided x,y values for input
+ */
+
+class OneDimensionalLinearInterpolator : public Interpolator {
+public:
+  OneDimensionalLinearInterpolator(std::vector<double> x_data,
+                                   std::vector<double> y_data,
+                                   SYCLTargetSharedPtr sycl_target)
+      : Interpolator(x_data, y_data, sycl_target){
+
+        };
+
+  virtual void get_y(std::vector<double> &x_input,
+                     std::vector<double> &y_output) {
+    interpolate(x_input, y_output);
+  }
+
+protected:
+  virtual void interpolate(std::vector<double> &x_input,
+                           std::vector<double> &y_output) {
+
+    // testing error handling of interpolator
+    ErrorPropagate ep(m_sycl_target);
+    // sycl code
+    auto k_ep = ep.device_ptr();
+    ep.reset();
+    y_output = std::vector<double>(x_input.size());
+    sycl::buffer<double, 1> buffer_x_data(m_x_data.data(),
+                                          sycl::range<1>{m_x_data.size()});
+    sycl::buffer<double, 1> buffer_y_data(m_y_data.data(),
+                                          sycl::range<1>{m_y_data.size()});
+    sycl::buffer<double, 1> buffer_x_input(x_input.data(),
+                                           sycl::range<1>{x_input.size()});
+    sycl::buffer<double, 1> buffer_y_output(y_output.data(),
+                                            sycl::range<1>{y_output.size()});
+    sycl::buffer<double, 1> buffer_dy(dy.data(), sycl::range<1>{dy.size()});
+    const int m_x_data_size = m_x_data.size();
+    const int m_y_data_size = m_y_data.size();
```

Comment:
How about in the constructor.
```
NESOASSERT(m_x_data.size() <= m_y_data.size(), "<error to say input data size is missmatched>");
```
then use m_x_data.size() for the data extent everywhere.

Reply:
Fixed but instead of <= I changed to !=